diff --git a/AK/BumpAllocator.h b/AK/BumpAllocator.h
index 48f325d766..7937e21f90 100644
--- a/AK/BumpAllocator.h
+++ b/AK/BumpAllocator.h
@@ -10,6 +10,13 @@
 #include <AK/StdLibExtras.h>
 #include <AK/Types.h>
 #include <AK/kmalloc.h>
+
+// The VirtualAlloc broke the build on windows, so we need to just rely on mman to do this for us.
+#ifdef AK_OS_WINDOWS
+#    define AK_OS_WINDOWS_WAS_DEFINED
+#    undef AK_OS_WINDOWS
+#endif
+
 #if !defined(AK_OS_WINDOWS)
 #    include <sys/mman.h>
 #else
@@ -220,3 +227,10 @@ inline Atomic<FlatPtr> BumpAllocator<use_mmap, size>::s_unused_allocation_cache
 using AK::BumpAllocator;
 using AK::UniformBumpAllocator;
 #endif
+
+
+// The VirtualAlloc broke the build on windows, so we need to just rely on mman to do this for us.
+#ifdef AK_OS_WINDOWS_WAS_DEFINED
+#    undef AK_OS_WINDOWS_WAS_DEFINED
+#    define AK_OS_WINDOWS
+#endif
\ No newline at end of file
